src/test/java/com/thealgorithms/datastructures/heaps/LeonardoHeapTest.java

```diff
@@ -0,0 +1,100 @@
+package com.thealgorithms.datastructures.heaps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class LeonardoHeapTest {
+
+    @Test
+    public void testAddElement() {
+        LeonardoHeap<Integer> heap = new LeonardoHeap<>();
+        heap.addElement(5);
+        heap.addElement(3);
+        heap.addElement(8);
+
+        assertEquals(8, heap.removeElement()); // Max element should be 8
+    }
+
+    @Test
+    public void testRemoveElement() {
+        LeonardoHeap<Integer> heap = new LeonardoHeap<>();
+        heap.addElement(10);
+        heap.addElement(20);
+        heap.addElement(5);
+
+        assertEquals(20, heap.removeElement());
+        assertEquals(10, heap.removeElement());
+        assertEquals(5, heap.removeElement());
+    }
+
+    @Test
+    public void testAddElementStrings() {
+        LeonardoHeap<String> heap = new LeonardoHeap<String>();
```

Comment:
Why to do not write just:

`LeonardoHeap<String> heap = new LeonardoHeap<>();`

As `String `already used in type declaration, other part will be infered.